frontend/src/concepts/analyticsTracking/useTrackUser.ts

```diff
@@ -26,9 +29,13 @@ export const useTrackUser = (username?: string): [IdentifyEventProperties, boole
       return aId;
     };
 
-    computeAnonymousUserId().then((val) => {
-      setAnonymousId(val);
-    });
+    if (!userID) {
+      computeAnonymousUserId().then((val) => {
+        setAnonymousId(val);
+      });
+    } else {
+      setAnonymousId(userID);
+    }
```

Comment:
If `userID` is the anonymous ID, then you can init the anonymous ID as follows instead of this if condition:
```
const [anonymousId, setAnonymousId] = React.useState<string | undefined>(userID);
```

Furthermore, if there's no reason to make use of `userID` aside from it being the anonymous ID, then you can remove the `userID` return value as well as the conditional tracking event based on user ID in `segmentIOUtils`

Reply:
Tbh I am not sure about the latter as the segment library seems to do some magic by introspecting property names and acting accordingly.

Reply:
I suspect that the segment library is under the hood evaluating the name of the first argument's variable and setting things accordingly